### DISCUSSION TITLE
Transferring evaluation values from Comsol to Python

### DISCUSSION THREAD

**Discussion by MagAtOQ** [MPh-py/MPh#93] [General]
Dear MPh,

I hope all goes well for you.
I'm trying to process datas from COMSOL using Python.
I guess this is possible using MPh... but it doesn't work as expected.

Here is my code:

```
import mph
client = mph.start(cores=1)
model = client.load('mySimulFile.mph')

a=model.evaluate(model.evaluate('yDisp',dataset='AmplitudeResponseSolution1'))
```
the yDisp is the name of an evaluation node. The values are already present in the mySimulFile.mph 
The 'AmplitudeResponseSolution1' is the dataset that is called in the mySimulFile.mph to achieve the yDisp evaluation.
This is the dataset of a time dependant analysis counting 1001 time steps.

However, when running the code, the 'a' variable is an empty 1D array of size 1001. It is not populated with the results. How do I have to do to access the values of that array from MPh?

Thanks a lot for you support.

And all the best

With regards

MG

**Comment by john-hen** [comment]
Hi. I have no idea how you get "an empty 1D array of size 1001" with that code. If anything, I would expect you to get some kind of error. I obviously cannot run your code, as I don't have your model file, but Comsol should throw an exception when you run this:
```python
a=model.evaluate(model.evaluate('yDisp',dataset='AmplitudeResponseSolution1'))
```` 

Why do you call `evaluate` *twice*?

Also:
> the yDisp is the name of an evaluation node

It shouldn't be "the name of an evaluation node". It should be an "expression" (in the Comsol sense), as per the signature of [`Model.evaluate`](https://mph.readthedocs.io/en/1.1/api/mph.Model.html#mph.Model.evaluate):
```python
evaluate(expression, unit=None, dataset=None, inner=None, outer=None)
```

**Comment by MagAtOQ** [comment]
Dear John-hen,

you are actually right, htere was a small failure in the code I transferred to you. Here is the original code:
`a=model.evaluate('yDisp',dataset='AmplitudeResponseSolution1')
`
I understand the need of setting up an expression to evaluate something, but this is, actually my problem... let's say that I want to evaluated the y-displacement of a given point. I have then to load the entire dataset related to this specific expression, for each time step, and from there, extract the sub-dataset that I'm interested in, in this case, the information related to a given point? There is no way to set up an evaluation into comsol and to access it directly? Maybe via a java command?

I hope that it has been a bit more clear.

Thanks anyway for your support

With regards

MG

**Comment by john-hen** [comment]
> There is no way to set up an evaluation into comsol and to access it directly?

But that's exactly what `model.evaluate()` does: It lets you access the results of an expression evaluated by Comsol.

As to what that expression needs to be in order to get only results for a specific point, that's really a Comsol question. I vaguely remember you need to set up a point integration of some sort, in your Comsol model. Comsol has documentation on that, somewhere.

**Comment by MagAtOQ** [comment]
Ok! Indeed, I tested and it worked that way... 

For who is interested, here is "the" procedure:
in comsol :
- setting up a point probe to extract given quantity - in this case, y-displacement
- finding the name of the probe in the Expression menu - in my case, it was "point1"
in python MPh
- running the simulation 
- extracting the value using a=model.evaluate('point1',dataset='AmplitudeResponseSolution1')

All good! Thanks for your help.
